src/tribler/core/components/content_discovery/community/tests/test_content_discovery_community.py

```diff
@@ -293,13 +293,26 @@ async def test_popularity_search(self):
             self.torrent_metadata(1)(title="ubuntu torrent", infohash=random_infohash())
             self.torrent_metadata(1)(title="debian torrent", infohash=random_infohash())
 
-        notifier = Mock()
-        self.overlay(0).composition.notifier = {notifications.remote_query_results: notifier}
-        self.overlay(0).send_search_request(**{"txt_filter": "ubuntu*"})
+        async def assert_response_received_on_search(search_query: dict):
+            notifier = Mock()
+            self.overlay(0).composition.notifier = {notifications.remote_query_results: notifier}
+            self.overlay(0).send_search_request(**search_query)
 
-        await self.deliver_messages()
+            await self.deliver_messages()
+
+            notifier.assert_called()
+
+        # Basic search query
+        query1 = {"txt_filter": "ubuntu*"}
+        await assert_response_received_on_search(query1)
+
+        # Query with unexpected parameters like 'exclude_deleted'
```

Comment:
NIT: It tests a query with "expected" unexpected parameters. Perhaps it would be better to slightly revise the comment.

Reply:
Substituted `unexpected parameters` with `deprecated parameters`